### PR TITLE
[JENKINS-60866] Un-inline JavaScript from ReverseProxySetupMonitor

### PR DESCRIPTION
**Pull request by Wadeck**
See [JENKINS-60866](https://issues.jenkins-ci.org/browse/JENKINS-60866).

As I was playing with the rootURL and the reverse proxy monitor, I fall into a situation where my root URL does not contain the contextPath (manually set) and thus it told me that the configuration is not good. So I added a logic for adding a warning message when such situation occurs.

The existing test was not really testing stuff, so I added several ones.

At the same time, I did some cleanup. If some of the changes do not satisfy you, we can keep only the un-inlining of JavaScript.
I quickly look for potential usage of the doTest method, and found nothing. So this breaking change should not have any impact. _(it's especially a bad idea to call programmatically a web method, but anyway)_

<details>

<summary>Screenshot of the additional warning message</summary>

Screenshot-2020-04-05_12-54-02

</details>

💡 If you have a typo fix, or unquestionable change, please suggest+commit directly ;)

### Proposed changelog entries

* (internal) Remove inline resources from ReverseProxySetupMonitor view
* Add a specific warning when the Jenkins Root URL does not contain the contextPath.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [n/a] For dependency updates: links to external changelogs and, if possible, full diffs

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a JIRA issue should exist and be labeled as `lts-candidate`

### CODE REVIEW

File: core/src/main/resources/hudson/diagnosis/ReverseProxySetupMonitor/resources.js
Comment (fqueiruga) [inline core/src/main/resources/hudson/diagnosis/ReverseProxySetupMonitor/resources.js:1]: Arrow functions do not work on some browsers jenkins supports. Basically, adjuncts cannot use them or other ES6 features.

File: core/src/main/resources/hudson/diagnosis/ReverseProxySetupMonitor/resources.js
Comment (fqueiruga) [inline core/src/main/resources/hudson/diagnosis/ReverseProxySetupMonitor/resources.js:4]: Can we return here to avoid one extra level of nesting by skipping the else clause?

File: core/src/main/resources/hudson/diagnosis/ReverseProxySetupMonitor/resources.js
Comment (fqueiruga) [inline core/src/main/resources/hudson/diagnosis/ReverseProxySetupMonitor/resources.js:6]: love seeing use of data-attributes 👍

File: core/src/main/resources/hudson/diagnosis/ReverseProxySetupMonitor/resources.js
Comment (fqueiruga) [inline core/src/main/resources/hudson/diagnosis/ReverseProxySetupMonitor/resources.js:20]: I would suggest renaming the `.context-message` class to `.js-context-message`, to show is intent to be used by JS code and avoid a regression when refactoring CSS.

Comment (fqueiruga) [inline core/src/main/resources/hudson/diagnosis/ReverseProxySetupMonitor/resources.js:20]: Again, arrow functions should be swapped by normal anonymous `function () { }`s

**Review by fqueiruga — CHANGES_REQUESTED** [review]

**Comment by fqueiruga** [comment]
Just put some comments. As you know, I'm a huge fan of un-inlining JS code. Great initiative!

File: core/src/main/resources/hudson/diagnosis/ReverseProxySetupMonitor/resources.js
Comment (fqueiruga) [reply core/src/main/resources/hudson/diagnosis/ReverseProxySetupMonitor/resources.js:20]: Looks like I failed to pick an actual line :|

Comment (fqueiruga) [reply core/src/main/resources/hudson/diagnosis/ReverseProxySetupMonitor/resources.js:20]: I failed to pick an actual line. This comment was meant for the forEach loop

**Comment by Wadeck** [comment]
@fqueiruga Good point for the arrow, I also changed the rest ;)

File: core/src/main/resources/hudson/diagnosis/ReverseProxySetupMonitor/resources.js
Comment (Wadeck) [reply core/src/main/resources/hudson/diagnosis/ReverseProxySetupMonitor/resources.js]: It's a really good idea to js-xxx for stuff only used as tag for JavaScript without any CSS implication. Do we have a guideline for others?

**Comment by daniel-beck** [comment]
While discouraged, it's possible to configure reverse proxies to add or remove context paths.

The way the message in the screenshot is phrased may be misleading as well. Or is it shown if and only if the context path is the only problem?

**Comment by Wadeck** [comment]
> Or is it shown if and only if the context path is the only problem?

Yes. If the application is at `http://localhost:8080/jenkins`, but your root URL is `http://localhost:8080`, the message will be displayed. If your root URL is `http://127.0.0.1`, it will not be displayed.

File: core/src/main/resources/hudson/diagnosis/ReverseProxySetupMonitor/resources.js
Comment (fqueiruga) [inline core/src/main/resources/hudson/diagnosis/ReverseProxySetupMonitor/resources.js:22]: You left one arrow function here :)

**Review by fqueiruga — APPROVED** [review]

**Comment by fqueiruga** [comment]
What's the status of this one? Are there any blockers?

**Comment by Wadeck** [comment]
@fqueiruga No blockers, but missing second approval ;)

**Review by res0nance — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by res0nance** [comment]
@daniel-beck are you still planning on reviewing this?

**Comment by daniel-beck** [comment]
Lack of my review is nonblocking. Thanks for checking!

**Comment by res0nance** [comment]
Marking ready to merge will merge if no negative feedback in 24h

File: core/src/main/resources/hudson/diagnosis/ReverseProxySetupMonitor/message.properties
Comment (jsoref) [inline core/src/main/resources/hudson/diagnosis/ReverseProxySetupMonitor/message.properties:1]: This should have been `set-up` or `configuration`

**Pull request merged**
